### PR TITLE
pkg: separately exclude files from the old OpenSolaris closed-bins

### PR DESCRIPTION
**Pull request by richlowe**
This separates the notion of "Not on AArch64" and "Is a closed-binary" to make it easier to reason about.

@citrus-it  @hadfl @ptribble I'd mentioned this to y'all before.  Perhaps you'd want to take a look?  I think to do something similar in illumos-gate we'd perhaps want to factor out the other CPU modules to match Nehalem-EX and things like that, but for arm, for right now, I'd rather not.

The remaining inlined closed-bins are things I can't work out how to factor on package boundaries.  Most especially things like the vHCI operations, and fwflash plugins.  This is also why, at the moment, `labeld` and the trusted packages aren't `closed-only`.  My work in progress has a separated `trusted` toggle, which I'm still not sure if I prefer to bundling it up with closed.  Comments on these things are appreciated, too!

### CODE REVIEW

**Comment by richlowe** [comment]
I should also mention that I know that `$(FOO)$(BAR)` breaks `pkgfmt` (rather, makes the output ugly), but I can't do anything about that.  I didn't finish factoring packages to make that less bad.  I didn't factor the toshiba laptop hotkey acpi module out because I have no idea what to call the resulting package and -- really -- we should probably remove it upstream, which would just make life more complicated here not less if I factored it out.

File: usr/src/man/man5/Makefile [hunk 1]
```diff
@@ -198,11 +196,16 @@ _MANFILES=	Intro.5				\
 		ypserv.5			\
 		zoneinfo.5
 
+$(HAVE_CLOSED_BINS)MANFILES +=	\
+		ike.config.5	\
+		ike.preshared.5	\
+
+
```
Comment (hadfl) [inline usr/src/man/man5/Makefile:202]: stray newline

File: usr/src/pkg/Makefile [hunk 2]
```diff
@@ -172,15 +172,21 @@ PM_TRANSFORMS= common_actions publish restart_fmri facets defaults \
 PM_FINAL_TRANSFORMS= strip_dependinfo
 PM_INC= transforms manifests
 
+# Only deliver this file when using a specific java verion.
```
Comment (hadfl) [inline usr/src/pkg/Makefile]: typo: version

File: usr/src/pkg/manifests/system-library-iconv-legacy.p5m
Comment (ptribble) [inline usr/src/pkg/manifests/system-library-iconv-legacy.p5m:1]: Do we need to have these at all? I convinced myself nothing used them so deleted them completely from Tribblix.

Comment (richlowe) [reply usr/src/pkg/manifests/system-library-iconv-legacy.p5m:1]: We don't (though we should fix iconv(1) not to scan for them and pretend we do).  It's just better for me to remove them upstream than here.

**Comment by richlowe** [comment]
I force-pushed this (sorry) after merging arm64-gate.  This should deal with @ptribble's comment, since I deleted the legacy iconv bits upstream, and I fixed @hadfl's comments previously.  I think?

Is everyone happy?

**Review by hadfl — APPROVED** [review]

**Pull request closed without merging**